gooddata-sdk/gooddata_sdk/catalog/data_source/service.py

```diff
@@ -159,6 +399,25 @@ def scan_data_source(
         scan_request: CatalogScanModelRequest = CatalogScanModelRequest(),
         report_warnings: bool = False,
     ) -> CatalogScanResultPdm:
+        """Scan data source specified by its id and optionally by specified scan request.
+        CatalogScanResultPdm contains PDM and warnings. Warnings contain information about
+        columns which were not added to the PDM because their data types are not supported.
+        Additional parameter report_warnings can be passed to suppress or to report warnings.
+        By default warnings are returned but not reported to STDOUT. If you set report_warnings
+        to True, warnings are reported to STDOUT.
+
+        Args:
+            data_source_id (str):
+                Data Source identification string. e.g. "demo"
+            scan_request (CatalogScanModelRequest, optional):
+                Options for the Scan Request. Defaults to CatalogScanModelRequest().
+            report_warnings (bool, optional):
+                Switch to turn on warnings. Defaults to False.
+
+        Returns:
+            CatalogScanResultPdm:
+                TODO hkad98
```

Comment:
An instance of CatalogScanResultPdm. Containing pdm itself and a list of warnings that occurred during scanning.